c7n/resources/s3.py

```diff
@@ -2600,6 +2600,78 @@ def process_bucket(self, bucket):
             Bucket=bucket['Name'], LifecycleConfiguration={'Rules': config})
 
 
+@filters.register('bucket-encryption')
+class BucketEncryption(Filter):
+    """Filters for S3 buckets that have bucket-encryption
+
+    :example
+
+        .. code-block: yaml
+
+            policies:
+              - name: s3-bucket-encryption-AES256
+                resource: s3
+                region: us-east-1
+                filters:
+                  - type: bucket-encryption
+                    crypto: AES256
+              - name: s3-bucket-encryption-KMS
+                resource: s3
+                region: us-east-1
+                filters
+                  - type: bucket-encryption
+                    crypto: aws:kms
+                    key: 1234abcd-12ab-34cd-56ef-1234567890ab
+
+    """
+    schema = type_schema('bucket-encryption',
+                         required=['crypto'],
+                         crypto={'type': 'string', 'enum': ['AES256', 'aws:kms']},
+                         key={'type': 'string'})
+
+    permissions = ('s3:GetBucketEncryption',)
+
+    def process(self, buckets, event=None):
+        results = []
+        with self.executor_factory(max_workers=2) as w:
+            futures = {w.submit(self.process_bucket, b): b for b in buckets}
+            for future in as_completed(futures):
+                b = futures[future]
+                try:
+                    if future.result():
+                        results.append(b)
+                except Exception as e:
+                    self.log.error("Message: %s Bucket: %s", e,
+                                   b['Name'])
+        return results
+
+    def process_bucket(self, b):
+        try:
+            client = bucket_client(local_session(self.manager.session_factory), b)
+            be = client.get_bucket_encryption(Bucket=b['Name'])
+            b['c7n:bucket-encryption'] = be
+        except ClientError as e:
+            if e.response['Error']['Code'] != 'ServerSideEncryptionConfigurationNotFoundError':
+                raise
+
+        rules = be.get('ServerSideEncryptionConfiguration', []).get('Rules', [])
+
+        for sse in rules:
+            if self.filter_bucket(sse):
+                return True
+
+    def filter_bucket(self, sse):
+        crypto = self.data.get('crypto', 'AES256')
+        key = self.data.get('key')
+        rule = sse.get('ApplyServerSideEncryptionByDefault')
+        algo = rule.get('SSEAlgorithm')
+
+        if crypto == 'AES256' and algo == 'AES256':
+            return True
+        elif crypto == 'aws:kms' and algo == 'aws:kms' and rule.get('KMSMasterKeyID') == key:
```

Comment:
it would be nice if we could do kms aliases here like we do with set, otherwise the usage construction gets awkward, as it becomes very account and region specific without the alias usage.

Reply:
sounds good.  working on that now.